File: src/main/org/nlogo/prim/etc/_randomstate.java
package org.nlogo.prim.etc ;

import org.nlogo.nvm.Reporter;
import org.nlogo.nvm.Syntax;

// only display the mainRNG state, the auxillary shouldn't matter
// since it doesn't affect the outcome of the model.

public final strictfp class _randomstate
	extends Reporter
{
	@Override
	public Object report( final org.nlogo.nvm.Context context )
	{
		return world.mainRNG.save() ;
	}
	@Override
	public Syntax syntax()
	{
		return Syntax.reporterSyntax( Syntax.TYPE_STRING ) ;
	}
}

File: src/main/org/nlogo/prim/etc/_randomstate.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim.etc

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Context, Reporter }

// only display the mainRNG state. the auxiliary shouldn't matter since it doesn't affect the
// outcome of the model.

class _randomstate extends Reporter {
  override def syntax =
    Syntax.reporterSyntax(Syntax.StringType)
  override def report(context: Context): String =
    world.mainRNG.save
}
